performance/src/main/java/org/mule/test/SorterSmallAppBenchmark.java

```diff
@@ -0,0 +1,162 @@
+import org.mule.runtime.api.lock.LockFactory;
+import org.mule.runtime.api.store.ObjectStoreManager;
+import org.mule.runtime.config.internal.DependencyGraphLifecycleObjectSorter;
+import org.mule.runtime.config.internal.registry.SpringLifecycleObjectSorter;
+import org.mule.runtime.config.internal.registry.SpringRegistry;
+import org.mule.runtime.config.internal.resolvers.AutoDiscoveredDependencyResolver;
+import org.mule.runtime.config.internal.resolvers.ConfigurationDependencyResolver;
+import org.mule.runtime.config.internal.resolvers.DeclaredDependencyResolver;
+import org.mule.runtime.config.internal.resolvers.DependencyGraphBeanDependencyResolver;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.Config;
+import org.mule.runtime.core.api.config.MuleConfiguration;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.security.SecurityManager;
+import org.mule.runtime.core.api.streaming.StreamingManager;
+import org.mule.runtime.core.api.util.queue.QueueManager;
+import org.mule.runtime.core.internal.context.MuleContextWithRegistry;
+import org.mule.runtime.core.internal.el.mvel.ExpressionLanguageExtension;
+import org.mule.runtime.core.internal.registry.MuleRegistry;
+import org.mule.runtime.extension.api.runtime.config.ConfigurationProvider;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+@Fork(1)
+@State(Scope.Benchmark)
+@OutputTimeUnit(MILLISECONDS)
+public class SorterSmallAppBenchmark extends FunctionalTestCase {
+
+  @Inject
+  private MuleContext muleContext;
+  private ConfigurationDependencyResolver configurationDependencyResolver;
+  private DeclaredDependencyResolver declaredDependencyResolver;
+  private AutoDiscoveredDependencyResolver autoDiscoveredDependencyResolver;
+  private Map<String, Object> lookupObjects;
+  private List<String> lookupObjectNames;
+  private DependencyGraphBeanDependencyResolver resolver;
+  private DependencyGraphLifecycleObjectSorter graphSorter;
+  private SpringLifecycleObjectSorter springSorter;
+  private final Class<?>[] allowedTypes = new Class<?>[] {
+      LockFactory.class,
+      ObjectStoreManager.class,
+      ExpressionLanguageExtension.class,
+      ExpressionLanguage.class,
+      QueueManager.class,
+      StreamingManager.class,
+      ConfigurationProvider.class,
+      Config.class,
+      SecurityManager.class,
+      FlowConstruct.class,
+      MuleConfiguration.class,
+      Initialisable.class
+  };
+
+  @Override
+  protected String getConfigFile() {
+    return "benchmark/small-app.xml";
```

Comment:
can this and the large one be parameters on the same benchmark class?

Reply:
@elrodro83 Since we are extending the FunctionalTestCase, we can use 'getConfigFiles' instead to have two config files in one class. However, we need two muleContexts to get two different springRegistries for each app. This part was a bit tricky for me. Instead of using parameters, here is a link to a version that has only one benchmark class for both apps. Please let me know which version looks better! https://github.com/mulesoft/mule-integration-tests/commit/db9fa9bc68a47d1e8f5a2660b6e1e76dd00613be

Reply:
how about doing a base class with the reusable stuff and have each concrete class with a different config? `getConfigFiles` won't work becuase that is inteneded to do a single app with many config files, not an app per file.